app/controllers/concerns/hyrax/works_controller_behavior.rb

```diff
@@ -437,5 +439,21 @@ def concern_has_file_sets?
     def uploaded_files
       UploadedFile.find(params.fetch(:uploaded_files, []))
     end
+
+    def available_admin_sets
+      admin_set_results = Hyrax::AdminSetService.new(self).search_results(:deposit)
+      # get all the templates at once, reducing query load
+      templates = PermissionTemplate.where(id: admin_set_results.map(&:id)).to_a
```

Comment:
Not a useful comment but this is a great optimization

Reply:
i'm glad you commented, because i think we probably want to get a controller test in that ensures this is working right.

we want this working reasonably for the following cases:

  - no admin sets exist;
  - one or more admin sets exist, and they all have permission templates;
  - one or more admin sets exist, and at least one is missing a permission template;
  - one or more admin sets exist, and all are missing permission templates.